src/Fetch.ml

```diff
@@ -379,6 +381,48 @@ module Response = struct
   external clone : t = "clone" [@@bs.send.pipe: t]
 end
 
+module FormData = struct 
+  module EntryValue = struct
+    type t
+
+    let classify t : [> `String of string | `File of file] =
+      if Js.typeof t = "string" then `String (Obj.magic t)
+      else `File (Obj.magic t)
+  end
+
+  module Iterator = Fetch__Iterator
+  type t = formData
+
+  external make : unit -> t = "FormData" [@@bs.new]
+  external append : string -> string -> unit = "append" [@@bs.send.pipe : t]
+  external delete : string -> unit = "delete" [@@bs.send.pipe : t]
+  external get : string -> EntryValue.t option = "get" [@@bs.send.pipe : t]
+  external getAll : string -> EntryValue.t array = "getAll" [@@bs.send.pipe : t]
+  external set : string -> string -> unit = "set" [@@bs.send.pipe : t]
+  external has : string -> bool = "has" [@@bs.send.pipe : t]
+  external keys : t -> string Iterator.t = "keys" [@@bs.send]
+  external values : t -> EntryValue.t Iterator.t = "values" [@@bs.send]
+
+  external appendObject : string -> _ Js.t -> ?filename:string -> unit =
```

Comment:
This is a bit nitpicky, but `<..> Js.t` is more accurate than `_ Js.t`. With this you could pass in an `int Js.t` for examplle, which would of course be silly, but what if at some point in the future BuckleScript decides to use `Js.t` to distinguish JS and OCaml semantics of something other than objects?

Reply:
I don't think there will ever be any way to create a valid `int Js.t` :-) BuckleScript libs written by Hongbo are also using something similar: https://bucklescript.github.io/bucklescript/api/Js.Obj.html#VALkeys (`val keys : 'a Js.t -> string array`)

Reply:
Most of the functions there use `< .. > Js.t`, and Bob's not exactly known for his consistency. But mostly I just don't see any good reason to be deliberately vague. It's not a technical improvement, there could be issues with it in the future, and being vague doesn't generally make for better documentation.

Reply:
Fair enough, I'll change it :-)